go-controller/pkg/network-attach-def-controller/network_manager.go

```diff
@@ -0,0 +1,145 @@
+package networkAttachDefController
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/controller"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type networkManager interface {
+	EnsureNetwork(util.NetInfo)
+	DeleteNetwork(string)
+	Start() error
+	Stop()
+}
+
+func newNetworkManager(name string, ncm NetworkControllerManager) networkManager {
+	nc := &networkManagerImpl{
+		name:               fmt.Sprintf("[%s network manager]", name),
+		ncm:                ncm,
+		networks:           map[string]util.NetInfo{},
+		networkControllers: map[string]*networkControllerState{},
+	}
+	// this controller does not feed from an informer, networks are manually
+	// added to the queue for processing
+	config := &controller.Config[string]{
+		RateLimiter: workqueue.DefaultControllerRateLimiter(),
+		Reconcile:   nc.sync,
+		Threadiness: 1,
+	}
+	nc.controller = controller.NewController(
+		nc.name,
+		config,
+	)
+	return nc
+}
+
+type networkControllerState struct {
+	controller         NetworkController
+	stoppedAndDeleting bool
+}
+
+type networkManagerImpl struct {
+	sync.Mutex
+	name               string
+	controller         controller.Controller
+	ncm                NetworkControllerManager
+	networks           map[string]util.NetInfo
+	networkControllers map[string]*networkControllerState
+}
+
+func (nm *networkManagerImpl) Start() error {
+	return controller.StartControllersWithInitialSync(nm.syncAll, nm.controller)
+}
+
+func (nm *networkManagerImpl) Stop() {
+	controller.StopControllers(nm.controller)
+
+	for _, networkControllerState := range nm.networkControllers {
+		networkControllerState.controller.Stop()
+	}
+}
+
+func (nm *networkManagerImpl) EnsureNetwork(network util.NetInfo) {
+	nm.Lock()
+	defer nm.Unlock()
+	nm.networks[network.GetNetworkName()] = network
+	nm.controller.Reconcile(network.GetNetworkName())
+}
+
+func (nm *networkManagerImpl) DeleteNetwork(network string) {
+	nm.Lock()
+	defer nm.Unlock()
+	delete(nm.networks, network)
+	nm.controller.Reconcile(network)
+}
+
+func (nm *networkManagerImpl) sync(network string) error {
+	nm.Lock()
+	defer nm.Unlock()
+
+	startTime := time.Now()
+	klog.V(5).Infof("%s: sync network %s", nm.name, network)
+	defer func() {
+		klog.V(4).Infof("%s: finished syncing network %s, took %v", nm.name, network, time.Since(startTime))
+	}()
+
+	want := nm.networks[network]
+	have := nm.networkControllers[network]
+
+	// we will dispose of the old network if deletion is in progress or if
+	// configuration changed
+	dispose := have != nil && (have.stoppedAndDeleting || !have.controller.CompareNetInfo(want))
```

Comment:
I know `CompareNetInfo` existed before, but it doesn't read well, do you think it is worth renaming to something like `SameAs(anotherNetInfo)`

Reply:
will think about i a new name